cloudapi/insights/client_test.go

```diff
@@ -297,6 +355,96 @@ func TestClient_IngestRequestMetadatasBatch_ReturnsErrorWithFailingServerAndNonC
 	require.ErrorContains(t, err, testErr.Error())
 }
 
+func TestClient_IngestRequestMetadatasBatch_ReturnsErrorWithNoErrorAfterRetrySeveralTimes(t *testing.T) {
+	t.Parallel()
+
+	// Given
+	ser := &mockWorkingIngesterServer{maxAttemptsTillSuccess: 10}
+	lis := newMockListener(t, ser)
+
+	cfg := ClientConfig{
+		ConnectConfig: ClientConnectConfig{Dialer: newMockContextDialer(t, lis)},
+		TLSConfig:     ClientTLSConfig{Insecure: true},
+		RetryConfig: RetryConfig{
+			MaxAttempts:          20,
+			RetryableStatusCodes: "INTERNAL,UNAVAILABLE",
+		},
+	}
+	cli := NewClient(cfg)
+	require.NoError(t, cli.Dial(context.Background()))
+	data := RequestMetadatas{
+		{
+			TraceID:        "test-trace-id-1",
+			Start:          time.Unix(9, 0),
+			End:            time.Unix(10, 0),
+			TestRunLabels:  TestRunLabels{ID: 1337, Scenario: "test-scenario-1", Group: "test-group-1"},
+			ProtocolLabels: ProtocolHTTPLabels{URL: "test-url-1", Method: "test-method-1", StatusCode: 200},
+		},
+		{
+			TraceID:        "test-trace-id-2",
+			Start:          time.Unix(19, 0),
+			End:            time.Unix(20, 0),
+			TestRunLabels:  TestRunLabels{ID: 1337, Scenario: "test-scenario-2", Group: "test-group-2"},
+			ProtocolLabels: ProtocolHTTPLabels{URL: "test-url-2", Method: "test-method-2", StatusCode: 401},
+		},
+	}
+
+	// When
+	err := cli.IngestRequestMetadatasBatch(context.Background(), data)
+
+	// Then
+	require.NoError(t, err)
+	require.True(t, ser.batchCreateRequestMetadatasInvoked)
+	require.True(t, ser.dataUploaded)
+	require.Equal(t, uint(10), ser.attempts)
+}
+
+func TestClient_IngestRequestMetadatasBatch_ReturnsErrorWithErrorAfterRetrySeveralTimes(t *testing.T) {
```

Comment:
Same here. Not really sure what `ReturnsErrorWithErrorAfterRetrySeveralTimes` means. Shouldn't it be something more like `ReturnsErrorAfterExhaustingMaxRetryAttempts`, or something along these lines.

Reply:
Changed to `ReturnsErrorAfterExhaustingMaxRetryAttempts`; it is much better, thanks!